sql/core/src/test/scala/org/apache/spark/sql/execution/command/v2/CheckConstraintSuite.scala

```diff
@@ -0,0 +1,241 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.execution.command.v2
+
+import org.apache.spark.sql.{AnalysisException, QueryTest}
+import org.apache.spark.sql.connector.catalog.Table
+import org.apache.spark.sql.connector.catalog.constraints.Check
+import org.apache.spark.sql.execution.command.DDLCommandTestUtils
+import org.apache.spark.sql.internal.SQLConf
+
+class CheckConstraintSuite extends QueryTest with CommandSuiteBase with DDLCommandTestUtils {
+  override protected def command: String = "Check CONSTRAINT"
+
+  test("Nondeterministic expression -- alter table") {
+    withTable("t") {
+      sql("create table t(i double)")
+      val query =
+        """
+          |ALTER TABLE t ADD CONSTRAINT c1 CHECK (i > rand(0))
+          |""".stripMargin
+      val error = intercept[AnalysisException] {
+        sql(query)
+      }
+      checkError(
+        exception = error,
+        condition = "NON_DETERMINISTIC_CHECK_CONSTRAINT",
+        sqlState = "42621",
+        parameters = Map("checkCondition" -> "i > rand(0)"),
+        context = ExpectedContext(
+          fragment = "i > rand(0)",
+          start = 40,
+          stop = 50
+        )
+      )
+    }
+  }
+
+  test("Nondeterministic expression -- create table") {
+    Seq(
+      "create table t(i double check (i > rand(0)))",
```

Comment:
nit. It would be great if we can capitalize `CREATE TABLE` like the other test cases for consistency in this file.